src/target/meson.build

```diff
@@ -0,0 +1,206 @@
+# This file is part of the Black Magic Debug project.
+#
+# Copyright (C) 2023 1BitSquared <info@1bitsquared.com>
+# Written by Rafael Silva <perigoso@riseup.net>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+target_common_includes = include_directories('.')
+
+target_common_sources = files(
+	'adiv5.c',
+	'adiv5_jtag.c',
+	'adiv5_swd.c',
+	'gdb_reg.c',
+	'jtag_devs.c',
+	'jtag_scan.c',
+	'sfdp.c',
+	'spi.c',
+	'target.c',
+	'target_flash.c',
+	'target_probe.c',
+)
+
+# TODO: What is the purpose of this? blindly ported from Makefile
```

Comment:
`swdptap_generic.c` needs fixing as it's still broken, but the two of them exist to serve when there's no low-level (`ll`), ie bitbanging, routines for a platform to allow us to talk SWD and JTAG generically over some unknown and platform-dependant transport that implements the `jtag_proc_s` and `swd_proc_s` interfaces.

Reply:
Thanks for clearing that up